test/parallel/test-regress-GH-5727.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var net = require('net');
+
+net.Server().listen(common.PORT, function() {
+  assert.equal(this._connectionKey, '6::::' + common.PORT);
+  this.close();
+});
+
+// The first argument is a configuration object
+assert.throws(function() {
+  net.Server().listen({ port: -1 >>> 0 }, assert.fail);
```

Comment:
I think we have `common.fail()` for cases like this, instead of `assert.fail()`.